spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/sql/TestSelect.java

```diff
@@ -120,4 +120,42 @@ public void testMetadataTables() {
         ImmutableList.of(row(ANY, ANY, null, "append", ANY, ANY)),
         sql("SELECT * FROM %s.snapshots", tableName));
   }
+
+  @Test
```

Comment:
I think we are missing a few failure test cases:
1. Cannot specify both snapshot-id and as-of-timestamp
2. Cannot write from table at a specific snapshot
3. Cannot delete from table at a specific snapshot

Reply:
Ack. I agree that it'd be good to have such test cases. I'd point out though that none of the above should be supported even before this change, so if the test cases don't exist, they are existing holes.

Reply:
Added test cases for reading with both snapshot-id and as-of-timestamp, inserting into a table at a specific snapshot, and deleting from a table at a specific snapshot.

Reply:
These look good to me. Thanks for adding them!